File: BinaryTree/php/treeMin.php
<?php

declare(strict_types=1);

require_once __DIR__ . '/TreeNode.php';

function treeMin(?TreeNode $root): int
{
    if ($root === null) {
        return PHP_INT_MAX;
    }

    return min($root->val, treeMin($root->left), treeMin($root->right));
}

File: BinaryTree/go/treeMin.go
package binarytree

// TreeMin returns the smallest value in the tree (or maxInt for an empty tree).
func TreeMin(root *TreeNode) int {
	if root == nil {
		return maxInt
	}

	leftMin := TreeMin(root.Left)
	rightMin := TreeMin(root.Right)
	if leftMin < rightMin {
		rightMin = leftMin
	}
	if root.Val < rightMin {
		return root.Val
	}
	return rightMin
}
